### PR TITLE
Allow mysql schemas that contain either a json or a datatime field

### PR DESCRIPTION
**Pull request by weiznich**
Unfortunately we cannot add a test for json, as the behaviour there
diverges between mariadb and mysql.

### CODE REVIEW

File: diesel_cli/src/print_schema.rs [hunk 1]
```diff
@@ -144,6 +144,8 @@ fn mysql_diesel_types() -> HashSet<&'static str> {
 
     types.insert("TinyInt");
     types.insert("Tinyint");
+    types.insert("Datetime");
```
Comment (JohnTitor) [inline diesel_cli/src/print_schema.rs:147]: Does it make sense to add `TIMESTAMP` along with it?

**Comment by chusteven** [comment]
> Unfortunately we cannot add a test for json, as the behaviour there diverges between mariadb and mysql.

Curious what `diesel`'s stance on MySQL vs. MariaDB; I'm guessing `diesel` supports both?

**Comment by weiznich** [comment]
> Curious what diesel's stance on MySQL vs. MariaDB; I'm guessing diesel supports both?

It's more like we officially support MySQL. MariaDB just works due to the fact that it is mostly the same thing. (In fact I personally use MariaDB locally for testing as this is just that what is provided by my distro.)
In the long term we would like to have a separate backend for MariaDB (https://github.com/diesel-rs/diesel/issues/1882) as this would allow us to support some SQL features that are not supported for MySQL yet without causing issues (`RETURNING` support https://github.com/diesel-rs/diesel/issues/2535)

File: diesel_cli/src/print_schema.rs (hunk 1 above)
Comment (weiznich) [reply diesel_cli/src/print_schema.rs:147]: It's already there: https://github.com/diesel-rs/diesel/blob/b51e6584ac4c6dc4495010e6e2f282c6de327e1f/diesel_cli/src/print_schema.rs#L79

**Pull request merged**